### PR TITLE
int8/uint8 support

### PR DESCRIPTION
**Pull request by wozeparrot**
This was done in the style of the float16 stuff. I tested that this works on GPU and CPU and TORCH.

### TODO:

- [x] Check https://github.com/geohot/tinygrad/blob/master/datasets/preprocess_imagenet.py. need to download imagenet first
- [x] Maybe do cast tests between half and int8/uint8 types

LLVM and CLANG don't work for half for me, so I'm opening this to test with CI.

### CODE REVIEW

**Comment by geohot** [comment]
Hehe I knew this one was too easy. Can you make the preprocess_imagenet dataset work as a test of this?

File: test/test_dtype.py [hunk 1]
```diff
@@ -86,5 +86,168 @@ def test_half_matmul_upcast(self):
     assert c.dtype == dtypes.float32
     np.testing.assert_allclose(c.numpy(), [[1,2],[3,4]])
 
+  def test_int8_to_np(self):
```
Comment (geohot) [inline test/test_dtype.py:89]: Is there some function here to prevent copy and paste?

**Comment by wozeparrot** [comment]
yup, this was way easier than I thought it would be :)

Should I make preprocess_imagenet an external test or a normal one fine?

**Comment by geohot** [comment]
Oh just testing locally is fine.

**Comment by wozeparrot** [comment]
huh it seems like preprocess_imagenet just works, unless I'm testing it wrong.

**Comment by geohot** [comment]
Yea...it works, but it has to have uint8 dtype or the file is 4x too big.

**Comment by wozeparrot** [comment]
> Yea...it works, but it has to have uint8 dtype or the file is 4x too big.

I think it does? my file is 1.8gb, not sure what it's supposed to be.

**Comment by geohot** [comment]
xxd it

**Comment by geohot** [comment]
Are there any tests of negative numbers? Are uint8 and int8 actually different?

**Comment by geohot** [comment]
Easy $200! Will send.

**Pull request merged**